gmso/utils/units.py

```diff
@@ -79,12 +76,352 @@ def default_reg():
+        charge_exp = dim_info[0][0][1][1][
+            current_idx
+        ]  # charge has (current_mks) in it, so this is the hint of how many
+        return (
+            dims
+            * u.dimensions.charge ** (-1 * charge_exp)
+            * chargeSym**charge_exp
+        )
+
+    @staticmethod
+    def _dimensions_from_thermal_to_energy(dims):
+        """Take a set of dimensions and substitute in Symbol("energy") to replace temperature."""
+        symsStr = str(dims.free_symbols)
+        temp_inBool = np.all([dimStr in symsStr for dimStr in ["temperature"]])
+        if not temp_inBool:
+            return dims
+        energySym = Symbol(
+            "(energy)"
+        )  # create dummy symbol to replace in equation
+        dim_info = dims.as_terms()
+        temp_idx = np.where(
+            list(map(lambda x: x.name == "(temperature)", dim_info[1]))
+        )[0][0]
+        temp_exp = dim_info[0][0][1][1][
+            temp_idx
+        ]  # energy has 1/time**2 in it, so this is the hint of how many
+        return (
+            dims
+            / u.dimensions.temperature**temp_exp
+            * energySym ** (temp_exp)
+        )
+
+    @classmethod
+    def _get_output_dimensions(cls, dims, thermal_equivalence=False):
+        if str(dims) == "1":  # use string as all dims can be converted
+            return u.dimensionless
+        dims = cls._dimensions_to_energy(dims)
+        dims = cls._dimensions_to_charge(dims)
+        if thermal_equivalence:
+            dims = cls._dimensions_from_thermal_to_energy(dims)
+        return dims
+
+
+class ljUnitSystem:
+    """Use this so the empty unitsystem has getitem magic method."""
+
+    def __init__(self, reg: u.UnitRegistry):
+        self.registry = reg
+        self.name = "lj"
+
+    def __getitem__(self, item):
+        """Return dimensionless units unless angle."""
+        if item == "angle":
+            return u.Unit("degree")
+        return u.Unit("dimensionless")
+
+
+def get_parameter_dimension(parameter, dimension):
+    """Return a unit from the parameter in a given dimension."""
+    param_terms = parameter.units.expr.as_terms()
+    uStr = ""
+    for symbol, exp in zip(param_terms[-1], param_terms[0][0][1][1]):
+        outputDim = LAMMPS_UnitSystems._get_output_dimensions(
+            u.Unit(symbol).dimensions
+        )
+        if str(outputDim) == dimension:
+            uStr += f"{symbol}*"
+        elif (
+            str(outputDim) == "dimensionless" and dimension == "(energy)"
+        ):  # add mol to units of energy
+            uStr += f"{symbol}**{exp}*"
+        elif (
+            str(outputDim) == "dimensionless" and dimension == "(mass)"
+        ):  # add mol to mass amu
+            uStr += f"{symbol}**{exp}*"
+    return u.Unit(uStr[:-1])
+
+
+def write_out_parameter_and_units(parameter_name, parameter, base_unyts=None):
+    """Take a parameter and return a heading for the parameter and the units it should be in."""
```

Comment:
Similar to above, I think we should add `Parameters` and `Returns` sections for docstring.

Reply:
Done in [f21a6c2](https://github.com/mosdef-hub/gmso/pull/747/commits/f21a6c2b74d3bd172ccbc6d87ba3eec5773299a3).